core/src/test/java/io/questdb/cairo/CreateTableTest.java

```diff
@@ -283,107 +370,86 @@ public void testCreateTableLikeTableWithMaxUncommittedRowsAndCommitLag() throws
+                "t timestamp) timestamp(t) partition by MONTH");
+        assertQuery("a\ty\tt\n", "select * from tab", "create table tab ( like x)", "t");
+        SymbolParameters parameters = new SymbolParameters(symbolCapacity, false, false, null);
+        assertSymbolParameters("tab", 1, parameters);
     }
 
     @Test
-    public void testCreateTableAsSelectIndexSupportedColumnTypeAfterCast() throws Exception {
-        assertQuery(
-                "x\n" +
-                        "1\n",
-                "select * from tab",
-                "CREATE TABLE tab AS (" +
-                        "SELECT CAST(x as SYMBOL) AS x FROM long_sequence(1)" +
-                        "), INDEX(x)",
-                null,
-                true,
-                true,
-                true
-        );
+    public void testCreateTableLikeTableWithWALDisabled() throws Exception {
+        createWALNonWal(false);
     }
 
     @Test
-    public void testCreateTableAsSelectIndexSupportedColumnTypeAfterCast2() throws Exception {
-        assertQuery(
-                "x\n" +
-                        "1\n",
-                "select * from tab",
-                "CREATE TABLE tab AS (" +
-                        "SELECT CAST(x as STRING) AS x FROM long_sequence(1)" +
-                        "), CAST(x as SYMBOL), INDEX(x)",
-                null,
-                true,
-                true,
-                true
-        );
+    public void testCreateTableLikeTableWithWALEnabled() throws Exception {
+        createWALNonWal(true);
     }
 
     @Test
-    public void testCreateTableAsSelectIndexUnsupportedColumnType() throws Exception {
-        assertFailure(
-                "CREATE TABLE tab AS (" +
-                        "SELECT x FROM long_sequence(1)" +
-                        "), INDEX(x)",
-                0,
-                "indexes are supported only for SYMBOL columns: x"
-        );
+    public void testCreateTableParallel() throws Throwable {
+        assertMemoryLeak(() -> {
+            int threadCount = 2;
+            int tableCount = 100;
+            AtomicReference<Throwable> ref = new AtomicReference<>();
+            CyclicBarrier barrier = new CyclicBarrier(threadCount);
+
+            ObjList<Thread> threads = new ObjList<>(threadCount);
+            for (int i = 0; i < threadCount; i++) {
+                int threadId = i;
+                threads.add(new Thread(() -> {
+                    try {
+                        barrier.await();
+                        try (
+                                SqlCompiler compiler = new SqlCompiler(engine);
+                                SqlExecutionContextImpl executionContext = new SqlExecutionContextImpl(engine, 1, 1)
+                        ) {
+                            for (int j = 0; j < tableCount; j++) {
+                                compiler.compile("create table if not exists tab" + (threadId * tableCount + j) + " (x int)", executionContext);
```

Comment:
This test shouldn't use IF NOT EXISTS clause, as its name suggests:
```suggestion
                                compiler.compile("create table tab" + (threadId * tableCount + j) + " (x int)", executionContext);
```